### ISSUE TITLE
In a prow job creating deployment doesn't check the status before continuing to next deployment.

### ISSUE THREAD

**Issue by krasi-georgiev** [prometheus/prombench#114]
see difference in the logs running from cli vs running in a prow job.

in local the deployment waits until it is ready before continuing to the next one where in prow it just continues.
 This causes the same time shifting problem  in the metrics  between pr and the release deployments I tried to solve by adding the ready probe .

this log is missing in the prow jobs.
```
Request for 'applying deployment:prometheus-test-pr-4448' is in progress. Checking in 10s
```

## Command to run
```
./prombench gke resource apply -a $AUTH_FILE \
	-v ZONE:europe-west3-a -v PROJECT_ID:macro-mile-203600 -v CLUSTER_NAME:prow \
	-v PR_NUMBER:4448 -v RELEASE:v2.3.2 \
	-f components/prombench/manifests/benchmark
```
## From local cli.
```
2018/08/06 13:46:33 resource created - kind: Namespace, name: prombench-4448
2018/08/06 13:46:33 resource created - kind: ServiceAccount, name: loadgen-scaler
2018/08/06 13:46:33 resource created - kind: ServiceAccount, name: prometheus
2018/08/06 13:46:33 resource created - kind: ClusterRoleBinding, name: loadgen-scaler
2018/08/06 13:46:33 resource created - kind: ClusterRoleBinding, name: prometheus-4448
2018/08/06 13:46:34 resource created - kind: Deployment, name: fake-webserver
2018/08/06 13:46:34 Request for 'applying deployment:fake-webserver' is done!
2018/08/06 13:46:34 resource created - kind: Service, name: fake-webserver
2018/08/06 13:46:34 Request for 'applying service:fake-webserver' is done!
2018/08/06 13:46:34 resource created - kind: ConfigMap, name: prometheus-loadgen
2018/08/06 13:46:34 resource created - kind: Deployment, name: prometheus-loadgen-scaler
2018/08/06 13:46:34 Request for 'applying deployment:prometheus-loadgen-scaler' is in progress. Checking in 10s
2018/08/06 13:46:44 Request for 'applying deployment:prometheus-loadgen-scaler' is done!
2018/08/06 13:46:45 resource created - kind: Deployment, name: prometheus-loadgen-querier
2018/08/06 13:46:45 Request for 'applying deployment:prometheus-loadgen-querier' is in progress. Checking in 10s
2018/08/06 13:46:55 Request for 'applying deployment:prometheus-loadgen-querier' is done!
2018/08/06 13:46:55 resource created - kind: Service, name: prometheus-loadgen-querier
2018/08/06 13:46:55 Request for 'applying service:prometheus-loadgen-querier' is done!
2018/08/06 13:46:55 resource created - kind: ConfigMap, name: prometheus-test
2018/08/06 13:46:55 resource created - kind: Deployment, name: prometheus-test-pr-44482018/08/06 13:46:55 Request for 'applying deployment:prometheus-test-pr-4448' is in progress. Checking in 10s
2018/08/06 13:47:05 Request for 'applying deployment:prometheus-test-pr-4448' is in progress. Checking in 10s
2018/08/06 13:47:15 Request for 'applying deployment:prometheus-test-pr-4448' is in progress. Checking in 10s
2018/08/06 13:47:25 Request for 'applying deployment:prometheus-test-pr-4448' is in progress. Checking in 10s
2018/08/06 13:47:35 Request for 'applying deployment:prometheus-test-pr-4448' is in progress. Checking in 10s
2018/08/06 13:47:46 Request for 'applying deployment:prometheus-test-pr-4448' is done!
2018/08/06 13:47:46 resource created - kind: Service, name: prometheus-test-pr-4448
2018/08/06 13:47:46 Request for 'applying service:prometheus-test-pr-4448' is done!
2018/08/06 13:47:46 resource created - kind: Deployment, name: prometheus-test-v2-3-2
2018/08/06 13:47:46 Request for 'applying deployment:prometheus-test-v2-3-2' is in progress. Checking in 10s
2018/08/06 13:47:56 Request for 'applying deployment:prometheus-test-v2-3-2' is done!
2018/08/06 13:47:56 resource created - kind: Service, name: prometheus-test-v2-3-2
2018/08/06 13:47:57 Request for 'applying service:prometheus-test-v2-3-2' is done!
2018/08/06 13:47:57 resource created - kind: Deployment, name: node-exporter-prometheus-test-v2-3-2
2018/08/06 13:47:57 Request for 'applying deployment:node-exporter-prometheus-test-v2-3-2' is in progress. Checking in 10s
2018/08/06 13:48:07 Request for 'applying deployment:node-exporter-prometheus-test-v2-3-2' is done!
2018/08/06 13:48:07 resource created - kind: Deployment, name: node-exporter-prometheus-test-pr-4448
2018/08/06 13:48:07 Request for 'applying deployment:node-exporter-prometheus-test-pr-4448' is in progress. Checking in 10s
2018/08/06 13:48:17 Request for 'applying deployment:node-exporter-prometheus-test-pr-4448' is done!
2018/08/06 13:48:17 resource created - kind: DaemonSet, name: node-exporter-nodes
2018/08/06 13:48:17 resource created - kind: Service, name: node-exporter
2018/08/06 13:48:17 Request for 'applying service:node-exporter' is done!
```

## From a prow job

```
2018/08/06 10:21:40 resource created - kind: Namespace, name: prombench-4448
2018/08/06 10:21:40 resource created - kind: ServiceAccount, name: loadgen-scaler
2018/08/06 10:21:40 resource created - kind: ServiceAccount, name: prometheus
2018/08/06 10:21:41 resource created - kind: ClusterRoleBinding, name: loadgen-scaler
2018/08/06 10:21:41 resource created - kind: ClusterRoleBinding, name: prometheus-4448
2018/08/06 10:21:41 resource created - kind: Deployment, name: fake-webserver
2018/08/06 10:21:41 Request for 'applying deployment:fake-webserver' is done!
2018/08/06 10:21:41 resource created - kind: Service, name: fake-webserver
2018/08/06 10:21:41 Request for 'applying service:fake-webserver' is done!
2018/08/06 10:21:41 resource created - kind: ConfigMap, name: prometheus-loadgen
2018/08/06 10:21:41 resource created - kind: Deployment, name: prometheus-loadgen-scaler
2018/08/06 10:21:41 Request for 'applying deployment:prometheus-loadgen-scaler' is done!
2018/08/06 10:21:41 resource created - kind: Deployment, name: prometheus-loadgen-querier
2018/08/06 10:21:41 Request for 'applying deployment:prometheus-loadgen-querier' is done!
2018/08/06 10:21:41 resource created - kind: Service, name: prometheus-loadgen-querier
2018/08/06 10:21:41 Request for 'applying service:prometheus-loadgen-querier' is done!
2018/08/06 10:21:42 resource created - kind: ConfigMap, name: prometheus-test
2018/08/06 10:21:42 resource created - kind: Deployment, name: prometheus-test-pr-4448
2018/08/06 10:21:42 Request for 'applying deployment:prometheus-test-pr-4448' is done!
2018/08/06 10:21:42 resource created - kind: Service, name: prometheus-test-pr-4448
2018/08/06 10:21:42 Request for 'applying service:prometheus-test-pr-4448' is done!
2018/08/06 10:21:42 resource created - kind: Deployment, name: prometheus-test-v2-3-2
2018/08/06 10:21:42 Request for 'applying deployment:prometheus-test-v2-3-2' is done!
2018/08/06 10:21:43 resource created - kind: Service, name: prometheus-test-v2-3-2
2018/08/06 10:21:43 Request for 'applying service:prometheus-test-v2-3-2' is done!
2018/08/06 10:21:43 resource created - kind: Deployment, name: node-exporter-prometheus-test-v2-3-2
2018/08/06 10:21:43 Request for 'applying deployment:node-exporter-prometheus-test-v2-3-2' is done!
2018/08/06 10:21:43 resource created - kind: Deployment, name: node-exporter-prometheus-test-pr-4448
2018/08/06 10:21:43 Request for 'applying deployment:node-exporter-prometheus-test-pr-4448' is done!
2018/08/06 10:21:43 resource created - kind: DaemonSet, name: node-exporter-nodes
2018/08/06 10:21:43 resource created - kind: Service, name: node-exporter
2018/08/06 10:21:43 Request for 'applying service:node-exporter' is done!
```

**Comment by sipian** [comment]
I think this can be closed by #130

**Comment by krasi-georgiev** [comment]
thanks
